From 19abd15d7aca7bb2ddc80c606c6c6cffa5c688e7 Mon Sep 17 00:00:00 2001
From: James Kosin <jkosin@apache.org>
Date: Sat, 17 Mar 2012 01:54:16 +0000
Subject: [PATCH] OPENNLP-471: added min and max token counts to dictionary to
 hold token counts (for shortest and longest) stored in the dictonary.

git-svn-id: https://svn.apache.org/repos/asf/opennlp/trunk@1301853 13f79535-47bb-0310-9956-ffa450edef68
---
 .../opennlp/tools/dictionary/Dictionary.java  | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/opennlp-tools/src/main/java/opennlp/tools/dictionary/Dictionary.java b/opennlp-tools/src/main/java/opennlp/tools/dictionary/Dictionary.java
index 67f60dc06..ff7f236dc 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/dictionary/Dictionary.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/dictionary/Dictionary.java
@@ -92,6 +92,8 @@ public String toString() {
 
   private Set<StringListWrapper> entrySet = new HashSet<StringListWrapper>();
   private final boolean isCaseSensitive;
+  private int minTokenCount = 99999;
+  private int maxTokenCount = 0;
 
 
   /**
@@ -146,6 +148,24 @@ public Dictionary(InputStream in, boolean caseSensitive) throws IOException, Inv
    */
   public void put(StringList tokens) {
       entrySet.add(new StringListWrapper(tokens));
+      minTokenCount = Math.min(minTokenCount, tokens.size());
+      maxTokenCount = Math.max(maxTokenCount, tokens.size());
+  }
+  
+  /**
+   * 
+   * @return minimum token count in the dictionary
+   */
+  public int getMinTokenCount() {
+      return minTokenCount;
+  }
+  
+  /**
+   * 
+   * @return maximum token count in the dictionary
+   */
+  public int getMaxTokenCount() {
+      return maxTokenCount;
   }
 
   /**
